gossip/emitter.go

```diff
@@ -218,15 +218,15 @@ func (em *Emitter) addTxs(e *inter.Event, poolTxs map[common.Address]types.Trans
 		for _, tx := range txs {
 			// enough gas power
 			if tx.Gas() >= e.GasPowerLeft || e.GasPowerUsed+tx.Gas() >= maxGasUsed {
```

Comment:
move the first IF into the end of the loop (i.e. check other IFs first).

Reply:
this check is the most cheap, that's why gas is checked first